### PR TITLE
[Pavlo Turchyn]_uit-39_API integration

### PR DESCRIPTION
**Pull request by FabianMoretti**

### CODE REVIEW

File: src/app/home-page/components/home-page/home-page.component.ts [hunk 1]
```diff
@@ -17,38 +19,40 @@ export class HomePageComponent implements OnInit, OnDestroy {
...
       this.productsService.getWishList().subscribe(data => {
         this.wishList = data;
-      })];
+      }),
+
+      this.dataService.get('api/homepage').subscribe(data => {
+        this.slideShowImages = data.slideshow;
+      }),
 
+      this.dataService.get('api/homepage').subscribe(data => {
```
Comment (dverbovyi) [inline src/app/home-page/components/home-page/home-page.component.ts]: 3 requests api/homepage. You need only 1

Comment (dverbovyi) [reply src/app/home-page/components/home-page/home-page.component.ts]: also unsubscription is needed on component destroy

**Review by dverbovyi — CHANGES_REQUESTED** [review]

File: src/app/home-page/components/home-page/home-page.component.ts [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { IBanner } from 'src/app/shared/interfaces';
 import { IProduct } from 'src/app/shared/interfaces/product';
 import { KillswitchService } from '../../../core/services/killswitch.service';
 import { ProductsService } from 'src/app/core/services/products.service';
+import { DataService } from 'src/app/core/services/data.service';
+import { environment } from 'src/environments/environment';
```
Comment (Roman-Klimkevych) [inline src/app/home-page/components/home-page/home-page.component.ts:9]: `environment` is imported but never used

File: src/app/home-page/components/home-page/home-page.component.ts [hunk 3]
```diff
@@ -17,38 +19,40 @@ export class HomePageComponent implements OnInit, OnDestroy {
...
   ) { }
 
   ngOnInit(): void {
     this.wishListEnabled = this.killswitchService.getKillswitch('wishListEnabled');
 
     this.subscriptions = [
-      this.productsService.getProducts().subscribe(data => {
-        this.products = data;
-        this.prepareRecentlyViewedItems();
+      this.dataService.get('api/homepage').subscribe(data => {
```
Comment (Roman-Klimkevych) [inline src/app/home-page/components/home-page/home-page.component.ts:40]: `'api/homepage' ` should not be hardcoded.

**Comment by dverbovyi** [comment]
a new one https://github.com/griddynamics/ui-intshp2-cp-okapi/pull/48

**Pull request closed without merging**